lighthouse-core/report/html/renderer/util.js

```diff
@@ -26,6 +26,40 @@ class Util {
     return PASS_THRESHOLD;
   }
 
+  static get MS_DISPLAY_VALUE() {
+    return `%10d${NBSP}ms`;
+  }
+
+  /**
+   * @param {string|Array<string|number>=} displayValue
+   * @return {string}
+   */
+  static formatDisplayValue(displayValue) {
+    if (typeof displayValue === 'string' || typeof displayValue === 'undefined') {
+      return displayValue || '';
+    }
+
+    const template = /** @type {string} */ displayValue.shift();
+    if (typeof template !== 'string') {
+      // First value should always be the format string, but we don't want to fail to build
+      // a report, return a placeholder.
+      return 'UNKNOWN';
+    }
+
+    let output = template;
+    while (displayValue.length) {
+      const replacement = /** @type {number|string} */ (displayValue.shift());
+      output = output.replace(/%([0-9.]+)?(d|s)/, match => {
```

Comment:
this allows more than one decimal point? and no support for granularity when it's a string?

Reply:
correct if you mess up your format string it'll put `NaN`
correct %s is just straight up replacement

Reply:
I meant should it be more like `/%([0-9]*(?:\.[0-9]*))?d|s/` or whatever (you covered the no granularity + string case now)

Reply:
sure, tweaked it a bit